misc/dbt-materialize/dbt/include/materialize/macros/relations.sql

```diff
@@ -0,0 +1,94 @@
+-- Copyright 2020 Josh Wills. All rights reserved.
+-- Copyright Materialize, Inc. and contributors. All rights reserved.
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License in the LICENSE file at the
+-- root of this repository, or online at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+-- Most of this macro is direct copies of the PostgreSQL counterpart.
+-- Here we have included a minor change to fix the following error:
+-- column "referenced_schema.name" must appear in the GROUP BY clause or be used in an aggregate function
```

Comment:
I'm really surprised the query worked in PostgreSQL but not in Materialize! Is that a bug in Materialize?

Reply:
Sorry, I've missed that comment here. 

Indeed it might be, I just tested the query as it is from the original macro [here](https://github.com/dbt-labs/dbt-postgres/blob/main/dbt/include/postgres/macros/relations.sql) and it works with Postgres but throws this error in Materialize:

```
ERROR:  column "referenced_schema.name" must appear in the GROUP BY clause or be used in an aggregate function
```

Reply:
Only `referenced_schema.name` and `dependent_schema.name` are affected, so it seems like we're struggling with the logic processing order for `GROUP BY` under specific scenarios.